File: 1137-height-checker/solution.go
import (
    "slices"
)

func heightChecker(heights []int) int {
    originHeights := make([]int, 0, len(heights))
    for _, h := range heights {
        originHeights = append(originHeights, h)
    }

    slices.Sort(heights)

    count := 0
    for i, v := range heights {
        if originHeights[i] != v {
            count++
        }
    }

    return count
}

File: 1137-height-checker/solution.py
class Solution:
    def heightChecker(self, heights: List[int]) -> int:
        n = len(heights)
        counting_sort = [0] * (max(heights) + 1)
        for height in heights:
            counting_sort[height] += 1
        
        correct_data = []
        for i, count in enumerate(counting_sort):
            if count > 0:
                for j in range(count):
                    correct_data.append(i)

        count = 0
        for i in range(n):
            if correct_data[i] != heights[i]:
                count += 1
        return count
        
